### PR TITLE
SRCH-4962 Upgrade ruby version for search-gov to 3.3

### PR DESCRIPTION
**Pull request by krbhavith**
## Summary
- Upgrade ruby to 3.3 for search-gov
- Resolve deprecations
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

**Review by stevenbarragan — CHANGES_REQUESTED** [review]
There are a few things that still need to be addressed.

File: lib/es.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@ module ELK
     private
 
     def self.client_config(mode)
-      Rails.application.secrets.dig(:analytics, :elasticsearch, mode).freeze
+      Rails.application.config.secret_keys.dig(:analytics, :elasticsearch, mode).freeze
```
[the remark this answers is not in the document]
Comment (krbhavith) [reply lib/es.rb:44]: @luisgmetzger do we always have multiple `writers` in elasticsearch configuration. Thinking how to refactor the file based on the above comment

**Review by stevenbarragan — APPROVED** [review]
LGTM

**Review by stevenbarragan — APPROVED** [review]
:shipit:

**Review by stevenbarragan — APPROVED** [review]
🚀

**Review by stevenbarragan — APPROVED** [review]
:shipit:

**Pull request merged**